### PR TITLE
EOS-19965: [SW_UPGRADE][HOT_FIX]: GPG signature validation fix & other minor fixes

### PR DESCRIPTION
**Pull request by dmitrydb**
Convert explicitly GPG public key to OpenPGP format. I added some minor fixes and improvements

### CODE REVIEW

File: api/python/provisioner/commands/validator/validator.py [hunk 1]
```diff
@@ -661,6 +664,59 @@ class AuthenticityValidator(PathValidator):
...
+            armor_header = content[0]
+            armor_tail = content[-1]
+
+        # NOTE: we check that the armor header and armor tail in binary
+        #  representation exist in the first and the last line of
+        #  the pub key file.
+        if (ARMOR_HEADER.encode() in armor_header
+                and ARMOR_TAIL.encode() in armor_tail):
+            # NOTE: it means that provided public key is in ASCII Armor format
+            cmd = f"gpg --yes --dearmor {pub_key_path.resolve()}"
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/validator/validator.py:701]: @dmitrydb what if someone does that twice so the output file is already exists? will gpg replace it or fail?

File: api/python/provisioner/commands/validator/validator.py [hunk 2]
```diff
@@ -661,6 +664,59 @@ class AuthenticityValidator(PathValidator):
...
+            try:
+                # NOTE: by default gpg tool converts the given file to the file
+                #  with the same name + '.gpg' extension at the end.
+                #  Directory is the same
+                cmd_run(cmd, targets=local_minion_id(),
+                        fun_kwargs=dict(python_shell=True))
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/validator/validator.py:707]: I believe python_shell=True is not required here (no pipes or redirections), we need just path a list instead

File: api/python/provisioner/commands/validator/validator.py [hunk 3]
```diff
@@ -661,6 +664,59 @@ class AuthenticityValidator(PathValidator):
...
+            except Exception as e:
+                logger.error("Can't convert ASCII Armor GPG public key "
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/validator/validator.py:709]: could you mention path to pub key here?

**Review by andkononykhin2 — APPROVED** [review]
@dmitrydb Looks good. Thanks.

File: api/python/provisioner/commands/validator/validator.py (hunk 1 above)
Comment (dmitrydb) [reply api/python/provisioner/commands/validator/validator.py:701]: @andkononykhin2 yes, you are right, the same file will be replaced. I checked this several times. For that reason I used `--yes` option:
```
gpg --dearmor ./RPM-GPG-KEY-Seagate
File `./RPM-GPG-KEY-Seagate.gpg' exists. Overwrite? (y/N)
```

File: api/python/provisioner/commands/validator/validator.py (hunk 3 above)
Comment (dmitrydb) [reply api/python/provisioner/commands/validator/validator.py:708]: Yes, sure. I added given public key into error message output.

File: api/python/provisioner/commands/validator/validator.py (hunk 2 above)
Comment (dmitrydb) [reply api/python/provisioner/commands/validator/validator.py]: Done

**Pull request merged**